### PR TITLE
[ci skip] Doc-only enhancements for Hash

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods ::new and ::try_convert.

### CODE REVIEW

**Comment by nobu** [comment]
I mean that the argument will be returned as-is.

File: hash.c [hunk 1]
```diff
@@ -1895,12 +1895,18 @@ rb_check_hash_type(VALUE hash)
  *  call-seq:
  *    Hash.try_convert(obj) -> new_hash or nil
  *
+ *
```
Comment (nobu) [inline hash.c:1898]: Why an additional line?

File: hash.c [hunk 2]
```diff
@@ -1895,12 +1895,18 @@ rb_check_hash_type(VALUE hash)
...
  *  Returns the Hash object created by calling <tt>obj.to_hash</tt>:
  *    require 'csv' # => true
  *    row = CSV::Row.new(['Name', 'Age'], ['Bob', 45])
  *    row.respond_to?(:to_hash)  # => true
  *    Hash.try_convert(row) # => {"Name"=>"Bob", "Age"=>45}
  *
+ *  Returns the given <tt>obj</tt> if it is a Hash:
+ *    h = {}
+ *    h1 = Hash.try_convert(h)
+ *    h1.object_id == h.object_id # => true
```
Comment (nobu) [inline hash.c:1908]: `equal?` would be better than `object_id` comparison.

Comment (BurdetteLamar) [reply hash.c:1907]: Don 't agree.  Not everyone will recognize that equal? means identity.  But object_id is highly intuitive.

Comment (ioquatix) [reply hash.c:1907]: @BurdetteLamar  `#object_id` is not something we should encourage. If you write:

```
h1.equal?(h2) # identity comparison
```

maybe it's enough to be intuitive and informative too?

File: hash.c [hunk 3]
```diff
@@ -1745,36 +1745,46 @@ set_proc_default(VALUE hash, VALUE proc)
...
- *     h["c"].upcase!   #=> "GO FISH: C"
- *     h["d"]           #=> "Go Fish: d"
- *     h.keys           #=> ["c", "d"]
+ *  Returns a new empty Hash object.
  *
+ *  The initial default value and initial default proc for the new hash
+ *  depend on which form above was used.
+ *  See {Default Values}[#class-Hash-label-Default+Values].
+ *
+ *  If neither argument nor block given,
```
Comment (ioquatix) [inline hash.c:1757]: Is there a reason why this line is wrapped here?

**Comment by ioquatix** [comment]
This PR looks really great.

Thanks for your effort.

I will attempt to review and merge any documentation PR you link me to.

While some amount of discussion is required, it may be easier to make the smaller changes as requested by the core team so we can merge your work. Even if you still believe that there should be a change, I'd rather not see a small point hold up major improvements.

**Comment by ioquatix** [comment]
Looks great, I will merge once CI passes.

**Comment by nobu** [comment]
Thank you, @BurdetteLamar and @ioquatix.
And sorry to be lazy response.

**Pull request merged**